From 7eb6c1d4a927c54e20842d7c8128130d70e7f058 Mon Sep 17 00:00:00 2001
From: Jason Lowe <jlowe@apache.org>
Date: Fri, 24 Apr 2015 22:02:53 +0000
Subject: [PATCH 1392/2848] YARN-3537. NPE when NodeManager.serviceInit fails
 and stopRecoveryStore invoked. Contributed by
 Brahma Reddy Battula (cherry picked from commit
 5e093f0d400f82f67d9b2d24253c79e4a5abacf9)

(cherry picked from commit cf4154676b892a36fe977c115bac52f9dabcc128)

Change-Id: Iab77e9baf6fc3f06e84ad73bbcd43af6f8dc695c
---
 .../yarn/server/nodemanager/NodeManager.java       |   26 +++++++++++---------
 1 file changed, 14 insertions(+), 12 deletions(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/NodeManager.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/NodeManager.java
index e9a7027..cf54949 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/NodeManager.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/NodeManager.java
@@ -163,18 +163,20 @@ private void initAndStartRecoveryStore(Configuration conf)
   }
 
   private void stopRecoveryStore() throws IOException {
-    nmStore.stop();
-    if (null != context) {
-      if (context.getDecommissioned() && nmStore.canRecover()) {
-        LOG.info("Removing state store due to decommission");
-        Configuration conf = getConfig();
-        Path recoveryRoot =
-            new Path(conf.get(YarnConfiguration.NM_RECOVERY_DIR));
-        LOG.info("Removing state store at " + recoveryRoot
-            + " due to decommission");
-        FileSystem recoveryFs = FileSystem.getLocal(conf);
-        if (!recoveryFs.delete(recoveryRoot, true)) {
-          LOG.warn("Unable to delete " + recoveryRoot);
+    if (null != nmStore) {
+      nmStore.stop();
+      if (null != context) {
+        if (context.getDecommissioned() && nmStore.canRecover()) {
+          LOG.info("Removing state store due to decommission");
+          Configuration conf = getConfig();
+          Path recoveryRoot =
+              new Path(conf.get(YarnConfiguration.NM_RECOVERY_DIR));
+          LOG.info("Removing state store at " + recoveryRoot
+              + " due to decommission");
+          FileSystem recoveryFs = FileSystem.getLocal(conf);
+          if (!recoveryFs.delete(recoveryRoot, true)) {
+            LOG.warn("Unable to delete " + recoveryRoot);
+          }
         }
       }
     }
-- 
1.7.9.5

